doc/BuildingWithCMake.md

```diff
@@ -54,6 +54,104 @@ CMake can generate build files for a number of backends. You can select the gene
+
+The following components of the build tools are recommended to be installed:
+
+ * Windows 10 SDK (>= 10.0.16299.0) for Desktop C++
+ * Visual C++ tools for CMake
+
+**Note:** 
+To use the `NMake` or `Ninja` generator with *Visual C++*, `cmake` must be run from a shell that can use 
+the compiler `cl` from the command line. To enable the shell, please run
+`C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build\vcvarsall.bat [platform]` where `platform`
+can be one of the following options: `x86_amd64`, `x86_arm uwp`, `x86_arm onecore`, `x64 8.1` or `x64 store 8.1`.
+
+**Note:**
+If you wish to build the `DDR` component, `CMake` should be able to find the path to the `DIA SDK`. There are two ways to specify
+the path:
+
+ * Create the `VSSDK140Install` environment variable and set its value to the path to the *Visual Studio Build Tools* installation,
+for example, `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools`.
+
+ * Create the `DIASDK` environment variable set its value to the path to the `DIA SDK` folder itself, for example,
+`C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\DIA SDK`.
+
+Then, to configure the build, let's do this:
+
+```bash
+# always prefer building outside the source tree
+mkdir -p build && cd build
+
+# configure the build using Ninja
+cmake -DCMAKE_CXX_COMPILER=cl -DCMAKE_C_COMPILER=cl .. -G"Ninja"
+
+# ... or using NMake
+cmake -DCMAKE_CXX_COMPILER=cl -DCMAKE_C_COMPILER=cl .. -G"NMake Makefiles"
+
+# ... and build it
+cmake --build .
+```
+
+### Configuring the build for using Clang
+
+*Clang* is released as part of regular *LLVM* releases. You can download the release versions from the
+[LLVM Download Page](http://releases.llvm.org/download.html). *Clang* attempts to be compatible with *MSVC* and provides a
+special version of compiler: `clang-cl`. A good news is here: the compiler can be used to build `OMR` for Windows.
+
+**Note:**
+`clang-cl` requires the same libraries and tools as `MSVC` does, so the
+[previous section](#configuring-the-build-for-using-visual-studio-build-tools) should be taken into account:
+*Visual Studio Build Tools* must be installed and `cmake` must be run from the *Visual Studio Command Prompt* (`vcvarsall.bat`).
+The `PATH` environment variable has to contain the `clang-cl` location.
+
+After preparing the environment, let's do this:
+
+```bash
+# always prefer building outside the source tree
+mkdir -p build && cd build
+
+# configure the build using Ninja
+cmake -DCMAKE_CXX_COMPILER=clang-cl -DCMAKE_C_COMPILER=clang-cl .. -G"Ninja"
+
+# ... or using NMake
+cmake -DCMAKE_CXX_COMPILER=clang-cl -DCMAKE_C_COMPILER=clang-cl .. -G"NMake Makefiles"
+
+# ... and build it
+cmake --build .
+```
+
+Please note that `Clang` is more rigorous than `MSVC` and generates much more warnings and occasionally even can not
+compile the code that correctly compiled by using its sibling from Microsoft.
+
 ### Major feature flags
```

Comment:
You've accidentally put the rest of this document under the "Configuring the build on Windows" heading. The rest of this document applies to every platform, and IMHO belongs at the beginning of the document. Can we move your new windows-specific instructions to the end of the document?

Reply:
I've changed the header level of the `Configuring the build on Windows` section and put it as a subsection of the `Configuring the build` section. You can see how it looks [here](https://github.com/samolisov/omr/blob/document-build-with-cmake-for-windows/doc/BuildingWithCMake.md#configuring-the-build-for-using-visual-studio-build-tools).

Reply:
Looks good, thanks.